cloud_tasks/http_queues/create_http_queue.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START cloud_tasks_create_http_queue]
+import urllib
+
+# HTTP Queues are currently in public beta
+from google.cloud import tasks_v2beta3 as tasks
+
+
+def create_http_queue(project: str, location: str, name: str, uri: str) -> tasks.Queue:
+    """Create an HTTP queue.
+    Args:
+        project: The project ID to create the queue in.
+        location: The location to create the queue in.
+        name: The ID to use for the new queue.
+        uri: The HTTP endpoint's URI for all tasks in the queue
+
+    Returns:
+        The newly created queue.
+    """
+
+    # Create a client.
+    client = tasks.CloudTasksClient()
+
+    # Create the HTTP Target for the queue. This property is required
+    # for an HTTP queue. For legacy reasons, this property is an object
+    # with multiple fields instead of a string containing a URI.
+
+    # Extract the various components of the URI provided by the caller
+    parsedUri = urllib.parse.urlparse(uri)
+
+    http_target = {
+        "uri_override": {
+            "host": parsedUri.hostname,
+            "uri_override_enforce_mode": 2,  # ALWAYS use this endpoint
+        }
+    }
+    if parsedUri.scheme == "http":  # defaults to https
+        http_target["uri_override"]["scheme"] = 1
```

Comment:
nit: can we substitute this by a constant (if one exists) to be more descriptive? Also if I read the proto file correctly `2` is used for HTTPS.

Reply:
Since it defaults to HTTPS, I only set it if the user wants HTTP instead. I replaced it with the constant tasks.types.UriOverride.Scheme.HTTP, which is a long way to say 1.